modules/bibdk_heimdal/includes/heimdalDatabase.inc

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * @file
+ * Datebase handling for bibdk_heimdal. When a user inserts an email
+ * a row in bibdk_heimdal_email_verify table is inserted for later
+ * verification by link
+ *
+ */
+
+class heimdalDatabase {
+  /**
+   * Get all rows from bibdk_heimdal_email_verify table
+   * @return mixed
+   */
+  public static function getAll(){
+    $rows = db_select('bibdk_heimdal_email_verify', 'b')
+      ->fields('b')
+      ->execute()
+      ->fetchAssoc();
+
+    return $rows;
+  }
+
+  /**
+   * Insert a row in bibdk_heimdal_email_verify table
+   *
+   * @param string $email
+   * @param array $agencies
+   * @return mixed (string)
+   *  uuid (primary key of table) to be used when generating an email link
+   */
+  public static function insert(string $email, array $agencies) {
+    $key = uuid_generate();
+    $libraries = serialize($agencies);
+    $date = time();
+
+    if(!self::checkUniqueMail($email)){
+      // email already exist - maybe user asked for another link? ..
+      // anyway delete it before insert
+      self::deleteByMail($email);
+    }
+
+    db_insert('bibdk_heimdal_email_verify')
+      ->fields(
+        array(
+          'uuid' => $key,
+          'libraries' => $libraries,
+          'email' => $email,
+          'date' => $date,
+        )
+      )->execute();
+
+    return $key;
+  }
+
+  /**
+   * Email is unique - check if it is
+   * @param $email
+   *   The email to check for
+   * @return bool
+   *   FALSE if user already exists
+   */
+  private static function checkUniqueMail($email){
+    $rows = db_select('bibdk_heimdal_email_verify', 'b')
```

Comment:
Consider this:

$result = db_select('table_name', 'table_alias')
            ->fields('table_alias')
            ->execute();
    $num_of_results = $result->rowCount();

Then you don't need to fetchAssoc - I think it is faster.